Sources/SpeziScheduler/Scheduler.swift

```diff
@@ -6,42 +6,35 @@
 // SPDX-License-Identifier: MIT
 //
 
-import Combine
 import Foundation
 import OSLog
 import Spezi
-import SpeziLocalStorage
 import SwiftUI
 import UIKit
 import UserNotifications
 
 
-/// The ``Scheduler/Scheduler`` module allows the scheduling and observation of ``Task``s adhering to a specific ``Schedule``.
+/// The ``Scheduler`` module allows the scheduling and observation of ``Task``s adhering to a specific ``Schedule``.
 ///
-/// Use the ``Scheduler/Scheduler/init(tasks:)`` initializer or the ``Scheduler/Scheduler/schedule(task:)`` function
-/// to schedule tasks that you can obtain using the ``Scheduler/Scheduler/tasks`` property.
-/// You can use the ``Scheduler/Scheduler`` as an `ObservableObject` to automatically update your SwiftUI views when new events are emitted or events change.
-public class Scheduler<Context: Codable>: NSObject, UNUserNotificationCenterDelegate, Module {
-    @Dependency private var localStorage: LocalStorage
-    
-    public private(set) var tasks: [Task<Context>] = [] {
-        didSet {
-            guard oldValue != tasks else {
-                return
-            }
-            
-            _Concurrency.Task {
-                await persistChanges()
-            }
-        }
-    }
+/// Use the ``Scheduler/init(prescheduleNotificationLimit:tasks:)`` initializer or the ``Scheduler/schedule(task:)`` function
+/// to schedule tasks that you can obtain using the ``Scheduler/tasks`` property.
+/// You can use the ``Scheduler`` as an `ObservableObject` to automatically update your SwiftUI views when new events are emitted or events change.
+public class Scheduler<Context: Codable>: NSObject, UNUserNotificationCenterDelegate, Module, LifecycleHandler, EnvironmentAccessible {
```

Comment:
Would it make sense to mark `Scheduler` as `@Observable` to ensure that updates are propagated into views?

Reply:
This question is partly answered within https://github.com/StanfordSpezi/SpeziScheduler/pull/33#discussion_r1393639559. 

Apart from that, there is state within `Scheduler` (Having TaskList being Observable by itself is enough to have views getting rerendered). `firstLaunch` is the only mutable property within Scheduler. But it is private-use only and therefore doesn't require observability (and wouldn't be obvseravle out of the box due to the `@AppStorage` property wrapper).